app/controllers/users/postmark_webhook_controller.rb

```diff
@@ -29,7 +29,7 @@ def accepted_webhook_types
     end
 
     def mark_email_bounced
-      BounceReport.where(email: params[:Email]).first_or_create!(
+      BounceReport.where(email: params[:Email]).create_or_find_by!(
```

Comment:
This won't work as intended.

Example:

```rb
# Succeeds on first try with one bounce type.
BounceReport.where(email: 'testbounce@example.com').create_or_find_by!(bounce_type: "TEST_1")

# Crashes with `ActiveRecord::RecordNotFound` when supplied another bounce type.
BounceReport.where(email: 'testbounce@example.com').create_or_find_by!(bounce_type: "TEST_2")
```

...as opposed to the original behaviour:

```rb
# Succeeds, returns a BounceReport with bounce_type TEST_1.
BounceReport.where(email: 'testbounce@example.com').first_or_create!(bounce_type: "TEST_1")

# Succeeds again, but returns a BounceReport with bounce_type TEST_1, and doesn't try to create one for TEST_2
BounceReport.where(email: 'testbounce@example.com').first_or_create!(bounce_type: "TEST_2")
```

There are major differences between how these two operate.

`find_or_create` will first attempt to find a record with only the conditions supplied to the scope prior to the `first_or_create` call. So only the `where` condition is applied when attempting to find the `first_` record. Since one is found, the `or_create` part of the method isn't run.

`create_or_find_by!` works quite differently.

For the second method call from my first example, both the `create` and the `or_find_by!` parts of this method, both the preceding scope and the hash values supplied to the method are used. In this case, it'll first try to create a record with the supplied `email` and `bounce_type`... fail because `RecordNotUnique` will be triggered by the unique index on the email column..., and then it'll try to find a record with both the `email` and `bounce_type`, but there is no record with with the `bounce_type` `"TEST_2"`, so it fails to find the record, and finally raises a `RecordNotFound`.

Reply:
Got it. I had realised this and removed the plain change from `first_or_create` to `create_or_find_by` from couple of other places in the PR. Though I missed this one. 
I have revoked this change.